client/main.js

```diff
@@ -146,11 +147,12 @@ Array.from( document.getElementsByClassName( 'wcc-root' ) ).forEach( ( container
 			</AppContainer>,
 			container
 		);
+		return createdStores[ routeClassName ];
 	});
 } );
 
-window.wcsGetAppStore = function( storeKey ) {
-	return createdStores[ storeKey ];
+window.wcsGetAppStoreAsync = function( storeKey ) {
```

Comment:
The dynamic loading functionality that was added in `1.24.0` means there's no guarantee that `createdStores` will be ready after the main js file is loaded. Returning a promise instead allows dependent code to wait for everything to be resolved first (see: https://github.com/woocommerce/woocommerce-admin/pull/5025/files#diff-8f89acda6e8c126c62f0d46b8d47dd73R276-R278)

Reply:
Renaming this will need to bump our minimum WC requirement to 4.5+. Should we do the requirement bump in this PR or make another issue for 1.24.2 TODO?

Reply:
I'm not sure if there's much we can do. WCS is still generally compatible with very old versions of WC core. Any version of WC with the prompt will always download the latest version of WCS and try to start it. `wcsGetAppStore` is broken as of `1.24.0`. I couldn't think of a way to make it backwards compatible `wcsGetAppStore` without trying to remove the code splitting.

Reply:
Good point on the prompt that always download the latest. Since this part of the code in WCS is only being used by wc-admin to get the latest WCS, I think this is fine as well.